src/image.jl

```diff
@@ -179,6 +179,17 @@ function write(f::FITS, data::Array{T};
     nothing
 end
 
+"""
+    write(hdu::ImageHDU, data::Array)
+
+Write data to an existing image HDU.
+"""
+function write(hdu::ImageHDU, data::Array{T}) where T
+    fits_assert_open(hdu.fitsfile)
+    fits_write_pix(hdu.fitsfile, ones(Int, length(size(data))), length(data), data)
```

Comment:
You might want to add a check that the dimensions of `data` match the dimensions of the data already in the HDU. I think you can use `sz = fits_get_img_size(hdu.fitsfile)` for this.